src/services/backend/fetching/FetchFavouriteSongs.js

```diff
@@ -0,0 +1,48 @@
+// service instance used to fetch a user's top Spotify tracks
+import {API_BASE_URL, STATUS_SUCCESS} from "../constants";
+import fetchWithData from "../../helpers/fetchWithData";
+import parseSpotifyTracks from "../../helpers/parseSpotifyTracks";
+
+class FetchFavouriteSongs {
+  constructor(user) {
+    this._status = STATUS_SUCCESS;
+    this._user = user;
+    this._results = [];
+    this._error = null;
+  }
+
+  perform() {
+    let favouriteSongsRequest = new Request(`${API_BASE_URL}/users/${this._user.username}/favouriteSongs`,
+      {
+        headers: {
+          "Authorization": this._user.secret,
+          "Session": this._user.sessionID
+        }
+      });
+
+    return fetchWithData(favouriteSongsRequest, this,
+      data => this._results = parseSpotifyTracks(this.results, data));
+  }
```

Comment:
you can use `memoize` here to cache the function result

Reply:
Like the search results, those are cached automatically and are only actually loaded once. I could, of course, simply store the results in a favorites array in the user state and only fetch if the array is empty. That would, however, mean that the user's favorites are never updated.